modules/system/sudo.nix

```diff
@@ -0,0 +1,15 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.system.sudo;
+in
+
+{
+  options = {
+    system.sudo.touchid.enable = mkEnableOption "Enable sudo authentication with Touch ID";
+  };
+
+  config = mkIf cfg.touchid.enable { system.patches = [ ./etc-pam.d-sudo.patch ]; };
```

Comment:
Sadly the idea of system.patches doesn't work as well as I initially thought. I couldn't find a way to re apply a patch or check if it's already applied. So currently this only runs once, meaning it doesn't work for files that change outside of rebuilds like system updates.

Reply:
Ah shoot, that will be a problem since macOS resets this file after every update. 

Given that `environment.etc` doesn't seem viable, since asking the user to delete/move the file in question seems like a recipe for trouble, I assume the only way forward here is via a custom activation script. I've got some ideas about how to implement that, but I just wanted to double check that I'm not missing some other more elegant way of implementing this option.

Reply:
Pushed a new commit with a first pass at implementing this using an activation script.

Reply:
I think it would be better to use another `sed` to delete the `pam_tid.so` line on disable, so that other changes to the `/etc` file since the `.orig` backup aren't wiped out.

A more general/scalable approach to this kind of thing could be to attach a comment containing some unique identifier to every block of text added to a system `/etc` file, and have centralized tooling for inserting and removing those blocks. This could also be used for `environment.shells` and similar options. Of course, this individual case is probably too simple to warrant infrastructure like that.

Reply:
> I think it would be better to use another sed to delete the pam_tid.so line on disable, so that other changes to the /etc file since the .orig backup aren't wiped out.

Good point

> A more general/scalable approach to this kind of thing could be to attach a comment containing some unique identifier to every block of text added to a system /etc file, and have centralized tooling for inserting and removing those blocks. This could also be used for environment.shells and similar options. Of course, this individual case is probably too simple to warrant infrastructure like that.

Totally agree this would be really nice to have, but don't currently have the time to play around with implementing right now.

I've update the script based on your feedback, it uses `sed` to delete the inserted like now, and plays around with the idea of using a unique identifier. Added a new thread the updated code block here: https://github.com/LnL7/nix-darwin/pull/228#discussion_r499774598